src/runtime/entries/azure-functions.ts

```diff
@@ -1,9 +1,28 @@
 import '#internal/nitro/virtual/polyfill'
 import { nitroApp } from '../app'
 
+const levels = ['log', 'info', 'warn', 'error']
+
+const higherOrderLog = (name: string, context) => {
+  const logFn = (params) => {
+    if (context[name]) {
+      context[name](...params)
+    } else if (context.log[name]) {
+      context.log[name](...params)
+    }
+  }
+
+  console[name] = logFn
+}
+/**
+ * intercepts all calls to console.* and redirects them to the azure context logger
+ */
+const interceptLogging = context =>
+  levels.forEach(m => higherOrderLog(m, context))
+
 export async function handle (context, req) {
   const url = '/' + (req.params.url || '')
-
+  interceptLogging(context)
```

Comment:
Will this have issues with logs being associated to the wrong request if there are multiple parallel requests?

If not (that is, if there is only one aggregated logging bucket), then do we need to run this on every incoming request or could we early return if the interceptor has already been run?

Reply:
As I understand it, it is a single logging bucket for the whole function app.

We could optimize and dont intercept ,if we are already intercepting.
The gains do not outweight the potentiell bugs, in my opinion, with the added complexity we get.

I rather intercept twice than miss a critical log.